File: frontend/src/api/verification.ts
import { useQuery, useMutation, useQueryClient } from '@tanstack/react-query';
import { fetchApi } from './client';

export interface VerificationItem {
  id: string;
  entity_id: string;
  entity_type: string;
  entity_name: string;
  claim_type: string;
  priority: 'high' | 'medium' | 'low';
  status: 'pending' | 'in_review' | 'verified' | 'rejected';
  assigned_to?: string;
  evidence_summary?: string;
  notes?: string;
  created_at: string;
  completed_at?: string;
}

export interface EntityResolutionLink {
  id: string;
  source_entity_id: string;
  target_entity_id: string;
  link_type: string;
  confidence: number;
  evidence: Record<string, any>;
  reviewer?: string;
  status: string;
  created_at: string;
}

export interface CorrectionRecord {
  id: string;
  entity_id: string;
  entity_type: string;
  field_name: string;
  old_value?: string;
  new_value: string;
  reason: string;
  reporter_email: string;
  status: string;
  created_at: string;
}

export function useVerificationQueue(statusFilter?: string) {
  const queryParam = statusFilter ? `?status=${statusFilter}` : '';
  return useQuery<VerificationItem[]>({
    queryKey: ['verification_queue', statusFilter],
    queryFn: () => fetchApi<VerificationItem[]>(`/api/v1/verification/queue${queryParam}`),
  });
}

export function useSignOffClaim() {
  const queryClient = useQueryClient();
  return useMutation({
    mutationFn: ({
      queueId,
      approved,
      notes,
    }: {
      queueId: string;
      approved: boolean;
      notes?: string;
    }) =>
      fetchApi<VerificationItem>(`/api/v1/verification/queue/${queueId}/sign-off`, {
        method: 'POST',
        body: JSON.stringify({ approved, notes }),
      }),
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ['verification_queue'] });
      queryClient.invalidateQueries({ queryKey: ['matches'] });
      queryClient.invalidateQueries({ queryKey: ['accounts'] });
    },
  });
}

export function useSubmitCorrection() {
  const queryClient = useQueryClient();
  return useMutation({
    mutationFn: (data: {
      entity_id: string;
      field_name: string;
      new_value: string;
      reason: string;
      old_value?: string;
    }) =>
      fetchApi<CorrectionRecord>('/api/v1/verification/corrections', {
        method: 'POST',
        body: JSON.stringify(data),
      }),
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ['verification_queue'] });
    },
  });
}

export function useEntityResolutionLinks() {
  return useQuery<EntityResolutionLink[]>({
    queryKey: ['entity_resolution_links'],
    queryFn: () => fetchApi<EntityResolutionLink[]>('/api/v1/verification/entity-resolution'),
  });
}

File: backend/app/models/verification.py
import uuid
from datetime import datetime, timezone
from typing import Optional, Dict, Any, List
from sqlalchemy import Column, String, Integer, Float, DateTime, Boolean, ForeignKey, Text, func
from sqlalchemy.dialects.postgresql import UUID, JSONB
from sqlalchemy.orm import relationship, Mapped, mapped_column
from app.models.base import Base

class VerificationQueue(Base):
    __tablename__ = "verification_queue"
    __table_args__ = {"schema": "gold"}

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    entity_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), nullable=False, index=True)
    entity_type: Mapped[str] = mapped_column(String(50), nullable=False) # company, person, signal, customs
    entity_name: Mapped[str] = mapped_column(String(255), nullable=False)
    claim_type: Mapped[str] = mapped_column(String(100), default="buyer_procurement_intent")
    priority: Mapped[str] = mapped_column(String(20), default="medium") # high, medium, low
    status: Mapped[str] = mapped_column(String(50), default="pending") # pending, in_review, verified, rejected
    assigned_to: Mapped[Optional[str]] = mapped_column(String(100), default="Trade OS Senior Research Analyst")
    evidence_summary: Mapped[Optional[str]] = mapped_column(Text)
    notes: Mapped[Optional[str]] = mapped_column(Text)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=func.now())
    completed_at: Mapped[Optional[datetime]] = mapped_column(DateTime(timezone=True), nullable=True)

class EntityResolutionLink(Base):
    __tablename__ = "entity_resolution_link"
    __table_args__ = {"schema": "gold"}

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    source_entity_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("silver.entity_company.id"), nullable=False)
    target_entity_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("silver.entity_company.id"), nullable=False)
    link_type: Mapped[str] = mapped_column(String(50), default="brand_subsidiary") # subsidiary, brand_subsidiary, duplicate, trade_alias
    confidence: Mapped[float] = mapped_column(Float, default=0.95)
    evidence: Mapped[dict] = mapped_column(JSONB, default=lambda: {"source": "German Commercial Register (Handelsregister)"})
    reviewer: Mapped[Optional[str]] = mapped_column(String(100), default="Entity Resolution Engine")
    status: Mapped[str] = mapped_column(String(50), default="confirmed")
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=func.now())

class CorrectionRecord(Base):
    __tablename__ = "correction_record"
    __table_args__ = {"schema": "gold"}

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    entity_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), nullable=False, index=True)
    entity_type: Mapped[str] = mapped_column(String(50), nullable=False) # company, person, signal
    field_name: Mapped[str] = mapped_column(String(100), nullable=False)
    old_value: Mapped[Optional[str]] = mapped_column(Text)
    new_value: Mapped[str] = mapped_column(Text, nullable=False)
    reason: Mapped[str] = mapped_column(Text, nullable=False)
    reporter_email: Mapped[str] = mapped_column(String(255), default="exporter_user@butlers.in")
    status: Mapped[str] = mapped_column(String(50), default="submitted") # submitted, applied, rejected
    reviewed_by: Mapped[Optional[str]] = mapped_column(String(100), nullable=True)
    reviewed_at: Mapped[Optional[datetime]] = mapped_column(DateTime(timezone=True), nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=func.now())
